ui/apps/platform/src/reducers/featureFlags.js

```diff
@@ -17,14 +17,49 @@ const featureFlags = (state = [], action) => {
     return state;
 };
 
+const error = (state = null, action) => {
+    switch (action.type) {
+        case types.FETCH_FEATURE_FLAGS.REQUEST:
+        case types.FETCH_FEATURE_FLAGS.SUCCESS:
+            return null;
+
+        case types.FETCH_FEATURE_FLAGS.FAILURE:
+            return action.error;
+
+        default:
+            return state;
+    }
+};
+
+const isLoading = (state = true, action) => {
+    // Initialize true for edge case before featureFlagSagas call fetchFeatureFlags.
```

Comment:
Is the default state of `isLoading=true` meant to signal that "we have not completed a request for the data yet"? 

Initially I thought that the idea of setting `isLoading` to `true` when a request hasn't been initialized yet was a little odd, but it makes sense if you think of the three possible states being
- A request to get the data has failed
- A request to get the data has succeeded
- We have not completed a request to get the data yet (regardless of what state the request is in)

The only potential downside is if we have logic somewhere that needs to determine whether or not a request has been launched yet, we won't be able to with a default setting of true here.

Reply:
Great point. Here is an example with 4 states:

1. https://redux.js.org/tutorials/essentials/part-5-async-logic#loading-state-for-requests

    ```ts
    status: 'idle' | 'loading' | 'succeeded' | 'failed'
    ```

2. Here is the best description that I found for the 3 states in https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/utils/fetchingReduxRoutines.ts#L1-L5

    https://dev.to/sunil12738/using-request-success-failure-pattern-in-redux-to-handle-async-actions-4d0#integrating-the-pattern-with-middlewares

    Example code (in article) dispatches request action to cause the request, but because the following sagas request directly without an action, I added the action to update state of `isLoading` reducer.
    * https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/sagas/authSagas.js#L223
    * https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/sagas/authSagas.js#L372
    * https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/sagas/featureFlagSagas.js#L14-L16